streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentUtils.java

```diff
@@ -16,78 +16,408 @@
+     * {@link StreamsConfig#RACK_AWARE_ASSIGNMENT_TRAFFIC_COST_CONFIG trafficCost}
+     * and {@link StreamsConfig#RACK_AWARE_ASSIGNMENT_NON_OVERLAP_COST_CONFIG nonOverlapCost}
+     * configs which balance cross rack traffic minimization and task movement.
+     * Setting {@code trafficCost} to a larger number reduces the overall cross rack traffic of the resulting
+     * assignment, but can increase the number of tasks shuffled around between clients.
+     * Setting {@code nonOverlapCost} to a larger number increases the affinity of tasks to their intended client
+     * and reduces the amount by which the rack-aware optimization can shuffle tasks around, at the cost of higher
+     * cross-rack traffic.
+     * In an extreme case, if we set {@code nonOverlapCost} to 0 and @{code trafficCost} to a positive value,
+     * the resulting assignment will have an absolute minimum of cross rack traffic. If we set {@code trafficCost} to 0,
+     * and {@code nonOverlapCost} to a positive value, the resulting assignment will be identical to the input assignment.
+     * <p>
+     * This method optimizes cross-rack traffic for standby tasks only. For active task optimization,
+     * use {@link #optimizeRackAwareActiveTasks}.
      *
      * @param kafkaStreamsAssignments the current assignment of tasks to KafkaStreams clients
      * @param applicationState        the metadata and other info describing the current application state
      *
-     * @return a new map containing the mappings from KafkaStreamsAssignments updated with the default
-     *         rack-aware assignment for standby tasks
+     * @return a new map containing the mappings from KafkaStreamsAssignments updated with the default rack-aware assignment for standy tasks
      */
     public static Map<ProcessId, KafkaStreamsAssignment> optimizeRackAwareStandbyTasks(
         final ApplicationState applicationState,
         final Map<ProcessId, KafkaStreamsAssignment> kafkaStreamsAssignments
     ) {
+        if (!hasValidRackInformation(applicationState)) {
+            LOG.warn("Cannot optimize standby tasks with invalid rack information.");
+            return kafkaStreamsAssignments;
+        }
+
+        final int crossRackTrafficCost = applicationState.assignmentConfigs().rackAwareTrafficCost();
+        final int nonOverlapCost = applicationState.assignmentConfigs().rackAwareNonOverlapCost();
+        final long currentCost = computeTaskCost(
+            applicationState.allTasks(),
+            applicationState.kafkaStreamsStates(false),
+            crossRackTrafficCost,
+            nonOverlapCost,
+            true,
+            true
+        );
+        LOG.info("Assignment before standby task optimization has cost {}", currentCost);
         throw new UnsupportedOperationException("Not Implemented.");
     }
 
+    private static long computeTaskCost(final Set<TaskInfo> tasks,
```

Comment:
This is literally only used to get the cost at the beginning of the method, right? (and only for logging purposes it seems? Weird but let's keep the original logic as usual) I know it's like this in the original implementation but as a small improvement, can we rename this method to `#computeInitialCost` or something like that which makes its purpose more clear?